File: EXAMPLES/ALPHA_LIFETIME/plot_alpha_lifetime.m
close all

% Plotting Style
fsize = 28;
lwidth = 4;

% Green
color_green = [18/255,173/255,42/255];

t_confined = load('alpha_life_time_gorilla.dat');
t_sample = logspace(-4,log10(max(t_confined)),100).';

confined_ratio = zeros(size(t_sample));
n_sampling = numel(t_confined);


for i = 1:numel(t_sample)
    L = t_confined >= t_sample(i);
    confined_ratio(i) = sum(L)/n_sampling;
end


% standard deviation
confined_sigma = sqrt(confined_ratio.*(1-confined_ratio) / n_sampling);



%% Plot 1: Confined ratio vs. time
fh = figure;
fh.Units = 'normalized';
fh.Position = [0.2995 0 0.5292 0.9037];%[0.3542 0.0787 0.4833 0.8250];

hold on
% Shading

 f6 = fill([t_sample;flipud(t_sample)],[confined_ratio-1.96.*confined_sigma;flipud(confined_ratio+1.96.*confined_sigma)],'k','linestyle','none');


 set([f6],'FaceAlpha',0.2);

h0 = plot(t_sample,confined_ratio);

hold off

set([h0],'LineWidth',lwidth)


h0.Color = 'r';
h0.LineStyle = '-';


xlabel('$t_\mathrm{trace}$ [s]','FontSize',fsize+2,'Interpreter','latex');
ylabel('$f_c$','FontSize',fsize+2,'Interpreter','latex');

a1 = gca;
a1.FontSize = fsize;
a1.XScale = 'log';
grid on
box on

l = legend(h0,'GORILLA Poly4, $N_{s}=N_{\vartheta}=N_{\varphi} = 101$');
l.Interpreter = 'latex';
l.Location = 'southwest';
l.FontSize = fsize+3;

t1 = title('$\mathbf{a}$: $t_\mathrm{trace} = 0.01$ s, $N_\mathrm{particles} = 10^4$');
t1.Interpreter ='latex';
t1.FontSize = fsize+2;

ylim([0.9,1])

File: EXAMPLES/ALPHA_LIFETIME/plot_alpha_lifetime.py
#!/usr/bin/env python3
"""
Plot confined alpha-particle fraction vs. tracing time.

Reads alpha_lifetime_gorilla.dat (one confinement time per line, in seconds)
produced by a prior run of gorilla_applets_main.x with i_option = 5.

Python translation of plot_alpha_lifetime.m.
"""

import numpy as np
import matplotlib as mpl
import matplotlib.pyplot as plt
mpl.rcdefaults()


t_confined = np.atleast_1d(np.genfromtxt('alpha_lifetime_gorilla.dat'))
n_sampling = t_confined.size

t_sample = np.logspace(-4, np.log10(t_confined.max()), 100)
confined_ratio = np.array([(t_confined >= ts).sum() / n_sampling for ts in t_sample])
confined_sigma = np.sqrt(confined_ratio * (1.0 - confined_ratio) / n_sampling)

fig, ax = plt.subplots(figsize=(10, 8))
ax.fill_between(t_sample,
                confined_ratio - 1.96 * confined_sigma,
                confined_ratio + 1.96 * confined_sigma,
                color='k', alpha=0.2, linewidth=0)
ax.plot(t_sample, confined_ratio, color='r', linewidth=2,
        label=r'GORILLA Poly4, $N_s = N_\vartheta = N_\varphi = 101$')

ax.set_xscale('log')
ax.set_xlabel(r'$t_\mathrm{trace}$ [s]')
ax.set_ylabel(r'$f_c$')
ax.set_ylim(0.9, 1.0)
ax.grid(True, which='both')
ax.set_title(r'$t_\mathrm{trace} = 0.01$ s, $N_\mathrm{particles} = 10^4$')
ax.legend(loc='lower left')

plt.tight_layout()
plt.savefig('alpha_lifetime.png', dpi=150)
plt.show()
